app/controllers/users/courses_controller.rb

```diff
@@ -7,7 +7,12 @@ class Users::CoursesController < ApplicationController
 
   def index
     @target = ALLOWED_TARGETS.include?(params[:target]) ? params[:target] : ALLOWED_TARGETS.first
-    course_names = Course.where(published: true).where(title: I18n.t("course_names.#{@target}")).pluck(:title)
+    course_names = if @target == 'other_courses'
+                     Course.where(published: false).pluck(:title)
+                   else
+                     Course.where(published: true).where(title: I18n.t("course_names.#{@target}")).pluck(:title)
+                   end
```

Comment:
```suggestion
    course_names =
      if @target == 'other_courses'
        Course.where(published: false).pluck(:title)
      else
        Course.where(published: true).where(title: I18n.t("course_names.#{@target}")).pluck(:title)
      end
```

if - elseの返り値を受け取る場合、こう言う感じで書くのがrubyのイディオム的なところがあります。